13_file_input_output/exercises/characters_in_file.c

```diff
@@ -0,0 +1,54 @@
+// Write a program that takes as command-line arguments a character and zero or more
+// filename. If no arguments follow the character, have the program read the standard
+// input. Otherwise, have it open each file in turn and report how many times the character
+// appears in each file. The filename and the character itself should be reported along with
+// the count. Include error-checking to see whether the number of arguments is correct and
+// whether the files can be opend. If a file can't be opened, have the program report that
+// fact and go on to the next file.
+#include <stdio.h>
+#include <stdlib.h>
+#include "../../11_strings/lib.h"
+
+#define SIZE 500
+
+int main(int argc, char * argv[])
+{
+    int i, ch;
+    int count = 0;
+    FILE * fp;
+    char str[SIZE];
+    if(argc > 2)
+    {
+        for(i = 2; i < argc; i++)
+        {
+            if((fp = fopen(argv[i], "r")) == NULL)
+            {
+                fprintf(stderr, "Can't open file \"%s\"\n", argv[i]);
+                continue;
+            }
+	    ch = getc(fp);
+            while(ch != EOF)
+            {
+                if(ch == argv[1][0])
+                    count++;
+		ch = getc(fp);
+            }
+            printf("File %s contains %d characters \"%c\"\n", argv[i], count, argv[1][0]);
+            fclose(fp);
+        }
+    }
```

Comment:
I have 2 files f1 and f2. f1 contains `foo bar buzz`. f2 contains `spam eggs ham`. The program output is 
```
./a.out a f1 f2
File f1 contains 1 characters "a"
File f2 contains 3 characters "a"
```